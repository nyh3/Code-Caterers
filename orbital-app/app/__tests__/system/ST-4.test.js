import { render } from '@testing-library/react-native';
import SavedPage from '../../test/saved3';
import * as responsiveScreen from 'react-native-responsive-screen';

jest.mock('react-native-responsive-screen', () => ({
  setWidth: jest.fn(),
  setHeight: jest.fn(),
  setFontScale: jest.fn(),
}));

responsiveScreen.setWidth(375);
responsiveScreen.setHeight(667);
responsiveScreen.setFontScale(1);

describe('Saved Page', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<SavedPage />);
    expect(toJSON()).toMatchSnapshot();
  });
});
