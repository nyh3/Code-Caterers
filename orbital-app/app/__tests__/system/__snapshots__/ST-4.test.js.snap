// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saved Page renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF5FA",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderBottomColor": "#DDD",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderBottomColor": "#2C0080",
          "borderBottomWidth": 2,
          "flex": 1,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Saved Menus
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
          "opacity": 1,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Saved Profiles
      </Text>
    </View>
  </View>
  <RCTScrollView
    style={
      {
        "flex": 1,
        "paddingTop": 20,
      }
    }
    testID="saved-menus-tab-indicator"
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            {
              "color": "#666",
              "fontSize": 16,
            }
          }
        >
          User has not saved any menu items.
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
